source/style-guide/style/ip-addresses.txt

```diff
@@ -4,37 +4,113 @@
 IP Addresses
 ============
 
-.. include:: /style-guide/includes/page-needs-update.rst
+.. default-domain:: mongodb
+
+.. include:: /includes/styles/corrections.rst
 
 An *IP address* uses a sequence of numbers to uniquely identify a
 particular computer on the Internet.
 
-When you're discussing IP addresses or referring to a specific IP
-address, don't use *IP* only; use *IP address*. You don't need to
-spell out *IP* on first use.
+- When discussing IP addresses or referring to a specific IP address,
+  don't use *IP* only; use *IP address*. You don't need to spell out
+  *IP* on first use.
+
+- When referring to a specific version of the IP, use *IPv4 address* or
+  *IPv6 address* as appropriate.
+
+Types of IP Addresses
+---------------------
+
+.. list-table::
+   :widths: 20 60 20
+   :header-rows: 1
+   :stub-columns: 1
+
+   * - Address Type
+     - Notation
+     - Separator
 
-When you need to refer to a specific version of the IP, use *IPv4
-address* or *IPv6 address* as appropriate.
+   * - IPv4
+     - four groups of one to three decimal digits
+     - periods (``.``)
+
+   * - IPv6
+     - eight groups of four hexadecimal digits
+     - colons (``:``)
+
```

Comment:
OK, love this doc, great work. I think right here would be the right place to talk briefly about public vs private IPs:

Knowing whether a public vs a private IP is appropriate for a given doc or example will hopefully be covered by the writer's knowledge, or the context provided in the ticket requesting the work. However, for cases where there is doubt, using the wrong one stands out starkly as a problem, and can sometimes lead to execution issues when trying to implement (long timeouts / hangs, accidentally affecting local network systems, etc).

Maybe something like:

> Public, Private, and Localhost Addresses
> 
> Both IPv4 and IPv6 addresses have special address ranges reserved for *private networks*: networks that are not generally-routable on the public internet, and usually reserved for local network usage (think your home network vs the greater internet at large). Generally, the feature or service you are using should carefully indicate if it functions differently on a public or private network, and you should use example IP addresses to match.
> 
> Here are the three private networks for the IPv4 address space: https://www.arin.net/reference/research/statistics/address_filters/
> 
> There is a third type of IPv4 network address called the *localhost* address, which is always `127.0.0.1`. This address is used to communicate only within the same server. Notably, this is the default IP address of a newly install `mongod` server, meaning it is not accessible to clients anywehre but on the same host (i.e. from a `mongo` shell on the same server).


Prooooobably too much?

Reply:
I added some content and context.